### ISSUE TITLE
app2.py

### ISSUE THREAD

**Issue by hellenowen** [hellenowen/app2#1]
from flask import Flask,render_template,session,request
from flask.ext.bootstrap import Bootstrap
from flask.ext.wtf import Form
from wtforms.fields import StringField,SubmitField
from wtforms.validators import Required

app=Flask(**name**)
bootstrap=Bootstrap(app)

class NameForm(Form):
    led=StringField('Which LED do you want to light on',validators=[Required()])
    submit=SubmitField('Light it')

@app.route('/',methods=['GET','POST'])
def index():
    form=NameForm()
    if form.validate_on_submit():
        #session['led']=form.led.data
        #return redirect(url_for('index'))
        #led=form.led.data
        led=1
    #return render_template('index.html',form=form,led=session.get('led'))
        return render_template('index.html',form=form,led=led)
    return render_template('index.html',form=form)

@app.errorhandler(404)
def page_not_found(e):
    return render_template('404.html'),404

@app.errorhandler(500)
def server_error(e):
    return render_template('500.html'),500

if **name**=="**main**":
    app.run()

**Comment by miguelgrinberg** [comment]
Can you add your templates to the repository? I think the errors are in the templates.

**Comment by hellenowen** [comment]
![problem](https://cloud.githubusercontent.com/assets/10844781/7552884/4d3e1014-f704-11e4-8985-f43936975a60.jpg)

like this.
I think why doesn't quick_form work is because of the wrong type of form.
And I also find that if there is no **init** in the defination of class NameForm,it will be not work ,too.

**Comment by hellenowen** [comment]
I will show you another problem.I use wtf.quick_form(form).
![an_problem](https://cloud.githubusercontent.com/assets/10844781/7552964/0bb61594-f707-11e4-85bf-77aaff9eae0f.jpg)

**Comment by miguelgrinberg** [comment]
The `quick_form` macro should be invoked as a function:

```
{{ wtf.quick_form(form) }}
```

**Comment by hellenowen** [comment]
Yes,I use it as a function.But there is an error——Internal Server Error.

**Comment by miguelgrinberg** [comment]
Enable debug mode `app.run(debug=True)` and you will get a stack trace.

**Comment by hellenowen** [comment]
And I'm confused about the type of form.Or there is some problems about StringField and SubmitField.I mean that form's type is not FormType.

**Comment by hellenowen** [comment]
Sorry,my English is poor.I deeply appreciate your patience.
